### DISCUSSION TITLE
設定ファイルの分割方法について

### DISCUSSION THREAD

**Discussion by mimikun** [nyaosorg/nyagos#438] [Q&A]
たとえばneovimやweztermのように設定ファイルを分割したいのですが、どのようにすればできるでしょうか？

例: 

### env.lua

```lua
local pnpm_home = xdg_config_home .. path_sep .. "pnpm"
nyagos.envadd("PNPM_HOME", nyagos.getenv("PNPM_HOME") or pnpm_home)
nyagos.envadd("PATH", nyagos.getenv("PNPM_HOME"))
```

### alias.lua

```lua
nyagos.alias.rm = "rm -i"
```

### .nyagos

```lua
require("env")
require("alias")
```

**Comment by hymkor** [comment]
設定ファイルについては Lua 5.1 の仕様どおりなので、neovim や wezterm とあまりかわりはないと思います。いただいた例ですと、おそらくはモジュールの検索パスが設定されていないだけだと思われますので、

```lua
package.path = "./?.lua;./?.lc;/usr/local/?/init.lua"
```
みたいに設定していただけばよいと思います。

require や package.path については Lua 側の機能で自分も詳しくないのですが、http://milkpot.sakura.ne.jp/lua/lua51_manual_ja.html#pdf-package.path が参考になると思います。

一応、こちらでも同様のファイルセットを作って試してみました。

#### ~/Share/Etc/env.lua

```lua
print("env.lua loaded")
local pnpm_home = xdg_config_home .. path_sep .. "pnpm"
nyagos.envadd("PNPM_HOME", nyagos.env.PNPM_HOME or pnpm_home)
nyagos.envadd("PATH", nyagos.env.PNPM_HOME)
```

#### ~/Share/etc/alias.lua

```lua
print("alias.lua loaded")
nyagos.alias.rm = "rm -i"
```

#### ~/.nyagos 

```lua
xdg_config_home = "(xdg_config_home)" -- ← nil でエラーがでないように
path_sep = ";" -- ← nil でエラーがでないように
package.path = nyagos.pathjoin(os.getenv("USERPROFILE"),"Share\\etc\\?.lua")

require("env")
require("alias")
```

これで起動時に
```
env.lua loaded
alias.lua loaded
```

が表示されており、alias へも反映されているので、一応期待どおり動くようです。

**Comment by mimikun** [comment]
ありがとうございます。
https://github.com/mimikun/dotfiles/commit/c0c7a0b4e4a08401736600adfd0241c8359d794c
早速設定しましたが概ね期待通り動作しています。
